File: test/language/expressions/python/FloatTypeTest.py
import Expressions

class FloatTypeTest(Expressions.TestCase):
    def testResult(self):
        floatTypeExpression = self.api.FloatTypeExpression(self.FLOAT_VALUE)
        result = self.FLOAT_VALUE * 2.0 + 1.0 / 0.5 > 1.0
        self.assertEqual(result, floatTypeExpression.result())

    FLOAT_VALUE = 15.5

File: test/language/expressions/cpp/FloatTypeTest.cpp
#include "math.h"

#include "gtest/gtest.h"

#include "expressions/float_type/FloatTypeExpression.h"

namespace expressions
{
namespace float_type
{

TEST(FloatTypeTest, result)
{
    FloatTypeExpression floatTypeExpression;
    const float floatValue = 15.0;
    floatTypeExpression.setFloatValue(floatValue);

    const bool result = (floatValue * 2.0f + 1.0f / 0.5f > 1.0f);
    ASSERT_EQ(result, floatTypeExpression.funcResult());
}

} // namespace float_type
} // namespace expressions
